### DISCUSSION TITLE
How to pass props to slots in parallel routes?

### DISCUSSION THREAD

**Discussion by asharifauzan** [vercel/next.js#58016] [Help]
Hello, I just created a layout that has 2 slots `@admin` and `@guest`. It works but I want to pass some data from the layout to rendered slots. I have been using React.cloneElement but it's not yet worked for me. 

layout.jsx
```jsx
'use client';

import React, { cloneElement } from 'react';

function layout({ guest, admin }) {
  const [mode, setMode] = React.useState('guest');

  const handleClick = () => {
    setMode((prevState) => (prevState === 'guest' ? 'admin' : 'guest'));
  };

  const Component = cloneElement(
    mode === 'admin' ? admin : guest,
    {
      myProps: 'hello world',
    },
  );

  return (
    <div>
      {Component}
      <button type="button" className="btn btn-primary" onClick={handleClick}>Toggle Role View</button>
    </div>
  );
}

export default layout;

```

output `@guest`:

![image](https://github.com/vercel/next.js/assets/52328248/63890ca7-e834-4a02-b162-50d59f10bf2c)

It's also occurs too `@admin` which it's not detected myProps that  I've passed in layout.jsx

**Comment by logstown** [comment]
Have you found a fix/workaround for this?

**Reply by asharifauzan** [reply]
nope, currently using component instead of slots in `_components` directory

**Reply by logemann** [reply]
Also interessted in this. I need to pass props from layout to the slot components.

**Reply by Haybuka** [reply]
I havent tried this yet, so im unsure of how it'll affect, what if you made use of context to pass said props to @slots.

**Reply by shyam-rockx** [reply]
Yup Context can be way to go, but would have to trade off to covert layout to client component.

**Comment by tiavina-mika** [comment]
Is there any workaround about this?

**Comment by cernockyd** [comment]
parallel routes seem ok for modal use cases, but for complex use cases with conditional page rendering etc, I would stick to classic components as of now
